kafka-transport-binding.md

```diff
@@ -0,0 +1,278 @@
+derived from the CloudEvent using a configured partitionKeyExtractor. The partitionKeyExtractor is a transport
+specific function that contains bespoke logic to extract and populate the value.
+
+### 3.2. Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.2.1. Content Type
+
+For the *binary* mode, the header `cloudEvents_contentType` property MUST be
+mapped directly to the CloudEvents `contentType` attribute.
+
+
+#### 3.2.2. Event Data Encoding
+
+The [`data` attribute](#21-data-attribute) byte-sequence MUST be used as the
+value of the Kafka message.
+
+#### 3.2.3. Metadata Headers
+
+All [CloudEvents][CE] attributes and
+[CloudEvent Attributes Extensions](primer.md#cloudevent-attribute-extensions)
+with exception of `data` MUST be individually mapped to and from the Header
+fields in the Kafka message.
+
+##### 3.2.3.1 Property Names
+
+CloudEvent attributes are prefixed with "cloudEvents_" for use in the
+[message-headers][Kafka-Message-Header] section.
+
+Examples:
+
+    * `eventTime` maps to `cloudEvents_eventTime`
+    * `eventID` maps to `cloudEvents_eventID`
+    * `cloudEventsVersion` maps to `cloudEvents_cloudEventsVersion`
+
+##### 3.2.4.2 Property Values
+
+The value for each Kafka header is constructed from the respective
+header's Kafka representation, compliant with the [Kafka message
+format][Kafka-Message-Format] specification.
+
+
+#### 3.2.5 Example
+
+This example shows the *binary* mode mapping of an event into the
+Kafka message. All other CloudEvents attributes
+are mapped to Kafka Header fields with prefix `cloudEvents_`.
+
+Mind that `cloudEvents_` here does refer to the event `data`
+content carried in the payload.
+
+``` text
+------------------ Message -------------------
+
+Topic Name: mytopic
+
+------------------- key ----------------------
+
+Key: mykey
+
+------------------ headers -------------------
+
+cloudEvents_contentType: application/avro
+cloudEvents_cloudEventsVersion: "0.1"
+cloudEvents_eventType: "com.example.someevent"
+cloudEvents_eventTime: "2018-04-05T03:56:24Z"
+cloudEvents_eventID: "1234-1234-1234"
+cloudEvents_source: "/mycontext/subcontext"
+       .... further attributes ...
+
+------------------- value --------------------
+
+            ... application data ...
+
+-----------------------------------------------
+```
+
+### 3.3. Structured Content Mode
+
+The *structured* content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple transports.
+
+#### 3.3.1. Kafka Content-Type
+
+The [Kafka `cloudEvent_contentType`] property field MUST be set to the media
+type of an [event format](#14-event-formats).
+
+Example for the [JSON format][JSON-format]:
+
+``` text
+content-type: application/cloudevents+json; charset=UTF-8
```

Comment:
This shows no prefix

Reply:
Is this missing a `ce_` or is the sentence in the first paragraph of this section incorrect?

Reply:
I think the intro paragraph is wrong